test/e2e/resize_nodes.go

```diff
@@ -259,27 +266,46 @@ func resizeRC(c clientset.Interface, ns, name string, replicas int32) error {
 	return err
 }
 
-func getMaster(c clientset.Interface) string {
-	master := ""
+// getMaster populates the externalIP, internalIP and hostname fields of the master.
+// If any of these is unavailable, it is set to "".
+func getMaster(c clientset.Interface) Address {
+	master := Address{}
+
+	// Populate the internal IP.
+	eps, err := c.Core().Endpoints(api.NamespaceDefault).Get("kubernetes")
+	if err != nil {
+		framework.Failf("Failed to get kubernetes endpoints: %v", err)
+	}
+	if len(eps.Subsets) != 1 || len(eps.Subsets[0].Addresses) != 1 {
+		framework.Failf("There are more than 1 endpoints for kubernetes service: %+v", eps)
+	}
+	master.internalIP = eps.Subsets[0].Addresses[0].IP
```

Comment:
can you actually check that it has a rfc 1918 prefix?  (10, 172.16, 192.168), or, if you don't think it's worth it please add a comment. If you check it, and it isn't a private ip, you should log an error and stick the ip in externalIP instead.  I don't think it matter which we trust if both this and --api-server turns out to be an external ip.